src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -343,3 +346,165 @@ class Config:
     @accepts(Int('id'))
     async def do_delete(self, id):
         return await self.middleware.call('datastore.delete', self._config.datastore, id)
+
+
+class RsyncTaskService(CRUDService):
+
+    class Config:
+        datastore = 'tasks.rsync'
+        datastore_prefix = 'rsync_'
+
+    async def validate_rsync_task(self, data):
+        verrors = ValidationErrors()
+
+        if ' ' in data.get("user"):
+            raise verrors.add("user", "Usernames cannot have spaces")
+
+        if data.get("extra"):
+                data["extra"] = " ".join(data["extra"])
+        else:
+            data["extra"] = ""
+
+        if data.get("month"):
+                if len(data['month']) == 12:
+                    data['month'] = '*'
+                else:
+                    data['month'] = ",".join(data['month'])
+
+        if data.get("dayweek"):
+                if len(data['dayweek']) == 7:
+                    data['dayweek'] = '*'
+                else:
+                    data['dayweek'] = ",".join(data['dayweek'])
+
+        mode = data.get("mode")
+        rmodule = data.get("remotemodule")
+        if mode == 'module' and not rmodule:
+            verrors.add('remotemodule', 'This field is required')
+
+        rpath = data.get("remotepath")
+        if mode == 'ssh' and not rpath:
+            verrors.add('remotepath', 'This field is required')
+
+        username = data.get("user")
+        if mode == 'ssh':
+            rport = data.get("remoteport")
+            if not rport:
+                verrors.add('remoteport', 'This field is required')
+  
+            try:
+                user = await self.middleware.call('user.query', [('username', '=', username)], {'get': True})
+            except IndexError:
+                verrors.add('user', 'Provided user does not exists')
+            else:
+                search = os.path.join(user['home'], ".ssh", "id_[edr]*.*")
+                if not glob.glob(search):
+                    verrors.add(
+                        "user",
+                        "In order to use rsync over SSH you need a user"
+                        "with a public key (DSA/ECDSA/RSA) set up in home dir."
```

Comment:
I've created an issue here: https://redmine.ixsystems.com/issues/27967
You may try fixing it together with this task (make it private key instead of public, also check for file permissions).